### ISSUE TITLE
Ranger, Rebel and Red Guard get different amount of XP for capturing buildings

### ISSUE THREAD

**Issue by xezon** [TheSuperHackers/GeneralsGamePatch#399]
Ranger, Rebel and Red Guard get different amount of XP for capturing buildings.

USA Ranger +15 XP
GLA Rebel + 12 XP
China Red Guard + 4 XP
China Black Lotus + ? XP

Investigate.

**Comment by commy2** [comment]
- Lotus gets 20 XP and needs 150 to promote to vet 1 (8 buildings).

- Ranger needs 40 XP to promote, so 3 buildings. With Advanced Training, he goes from vet 0 straight to vet 2 after capturing two buildings.

- Rebel also needs 40 XP, so 4 buildings.

- Red Guard only needs 20 XP to promote, so 5 buildings.

- Interestingly, the XP from capturing buildings is transferred to the player XP. The values are rather miniscule though. 3 captured oils as USA are not even worth 1 killed vehicle (45 vs 50).

**Comment by Jundiyy** [comment]
I don't think this is really a problem, yes it seems right to make them all even (and wouldn't hurt I suppose) but at the same time perhaps they were done like this on purpose, due to all the weird numbers. There isn't really a pattern.

I can't say for sure but presume it has to do with how the unit is. Price, health, damage, build time etc.

**Comment by commy2** [comment]
I don't see any reason to act here either. It's just a neat quirk that the basic infantry has some slight differences.

The only problem I see here is Lotus taking forever to vet up. 8 buildings will almost never happen, so it will almost never reach vet 1 let alone 2 or 3.

Edit: Cash Steal also gives her 20 XP. Disabling vehicles gives nothing.

**Comment by Jundiyy** [comment]
Lotus does manage to vet up in some games as she gets it from cash steal and stealing buildings. I suppose it is kind of fair for her to need more captures to vet up as she captures at around half the speed of the other units and also has range, which saves even more time, so might become 1/4 of the time. 

Perhaps not having xp for vehicle disable was done on purpose?
Maybe because it's common and easy, and would get xp really fast?

**Comment by xezon** [comment]
I think I have never seen a Vet 2 Lotus. It is a bit of shame. Burton and Jarmen have much easier time to vet up. Lotus gets nothing for captures most of the time, because players just sell their buildings before Lotus can capture. Is there any way to do something about it?

**Comment by Jundiyy** [comment]
A mistake I've seen many people make, I do mention it when I see it, (I suppose many still don't know), is that players stop the units from capturing if the opponent starts to sell. 
If you let your unit carry on capturing and it captures during the sell stage, unit will still gain xp.

Burton and Kell rank up faster because they are usually used for more damage, Burton blows stuff up and kills inf.
Kell also snipes and snipes.

**Comment by commy2** [comment]
Lotus XP curve is 150 450 900.

Each building hack or cash steal is 20 XP.

So you need 8 hacks for vet 1, a total of 23 hacks for vet 2, and a total of 45 hacks for vet 3.

No Lotus will ever reach vet 3 in a legit game.

**Comment by Madrageeeeee** [comment]
@Jundiyy although a nice little fact about the sell and xp gain for lotus. I would almost always prefer to get away quicker and ‘cloak up’ than getting that little xp boost. 

Also often you would use cash hack as on a supply in whicch case you’d let her capture on anyway. 

Also a vetted lotus is not that much better to have and will give your enemy more xp when killed which is very likely to happen once you start capturing. 

All in all dont think this is a masive thing and is low on the priority list.

**Comment by commy2** [comment]
Good point about XP droped by Lotus when killed. The values are:
50 @ vet 0 | 100 @ vet 1 | 150 @ vet 2 | 400 @ vet 3

So for some strange reason, Lotus is (one of) the only unit(s) that doubles XP drop when killed at vet 1, and at vet 3, it's 8 times the base value instead of 3 times.

E.g. Burton and Jarmen are:
50 @ vet 0 | 50 @ vet 1 | 100 @ vet 2 | 150 @ vet 3
so in line with almost all vehicles that are not tanks.

Edit:
We should investigate what the benefits are for Lotus when vetted up. More health as well healing at vet 2, but does the hacking actually become faster? I vaguely remember Lotus eating Jarmen and Pathfinder shots when vetted up more than the Jarmen/Pathfinder.

**Comment by Jundiyy** [comment]
Do agree with Rage there. Vet 2 would of course mean more health and self heal but it doesn't make her hack/disable faster or have more range. So you'd prob not want to give away too much xp. I do like that thought.
